showcase/src/app/shared/example-viewer/example-viewer.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { MatButtonModule, MatIconModule, MatTabsModule, MatTooltipModule } from "@angular/material";
+import * as hljs from "highlight.js";
+import { HighlightJsModule, HIGHLIGHT_JS } from "angular-highlight-js";
+
+import { ExampleViewerComponent } from "./example-viewer.component";
+import { ExampleViewerService } from "./example-viewer.service";
+import { throwError, of } from "rxjs";
+
+export function highlightJsFactory(): any {
+	return hljs;
+}
+fdescribe("ExampleViewerComponent", () => {
+	let component: ExampleViewerComponent;
+	let exampleViewerService: ExampleViewerService;
+	let fixture: ComponentFixture<ExampleViewerComponent>;
+
+	beforeEach(async(() => {
+		return TestBed.configureTestingModule({
+			declarations: [ExampleViewerComponent],
+			imports: [
+				BrowserAnimationsModule,
+				HttpClientModule,
+				HighlightJsModule.forRoot({
+					provide: HIGHLIGHT_JS,
+					useFactory: highlightJsFactory
+				}),
+				MatButtonModule,
+				MatIconModule,
+				MatTabsModule,
+				MatTooltipModule
+			],
+			providers: [{ provide: ComponentFixtureAutoDetect, useValue: true }, ExampleViewerService]
+		}).compileComponents();
+	}));
+
+	beforeEach(() => {
+		fixture = TestBed.createComponent(ExampleViewerComponent);
+		component = fixture.componentInstance;
+		exampleViewerService = fixture.debugElement.injector.get<ExampleViewerService>(ExampleViewerService as any);
+	});
+
+	describe("@Input() title", () => {
+		it("should change the title according to the @Input", () => {
+			const h2: any = fixture.nativeElement.querySelector(".example-viewer-title-spacer h2");
+			component.title = "Test title";
+			fixture.detectChanges();
+			expect(h2.textContent).toContain(component.title);
+		});
+	});
+
+	describe("example-viewer-service", () => {
+		it("should not do anything when the file doesn't exist", () => {
+			spyOn(component, "addFileContent");
+			spyOn(exampleViewerService, "fetchFileAlt").and.returnValue(throwError("http failure"));
+			component.fetchFiles();
+			expect(component.addFileContent).not.toHaveBeenCalled();
+			expect(exampleViewerService.fetchFileAlt).toHaveBeenCalledTimes(component.extensions.length);
+		});
+
+		it("should call the callback when the file exists", () => {
+			spyOn(component, "addFileContent");
+			spyOn(exampleViewerService, "fetchFileAlt").and.returnValue(of("test"));
+			component.fetchFiles();
+			expect(component.addFileContent).toHaveBeenCalledTimes(component.extensions.length);
+			expect(exampleViewerService.fetchFileAlt).toHaveBeenCalledTimes(component.extensions.length);
+		});
+	});
+
+	describe("toggleSourceView()", () => {
+		it("should toggle the showSource property", () => {
+			component.showSource = true;
+			component.toggleSourceView();
+			expect(component.showSource).toBe(false);
+			component.toggleSourceView();
+			expect(component.showSource).toBe(true);
+		});
+	});
+
+	describe("@Input() extensions", () => {
```

Comment:
Let's put all the describe blocks related to inputs at the top and the rest of tests after so we can have more or less a convention on how to structure our unit tests ;)